packages/network/src/index.ts

```diff
@@ -62,10 +66,35 @@ export class StacksMainnet implements StacksNetwork {
 
   readonly coreApiUrl: string;
 
-  constructor(networkUrl: NetworkConfig = { url: HIRO_MAINNET_DEFAULT }) {
-    this.coreApiUrl = networkUrl.url;
+  constructor(networkConfig: NetworkConfig) {
+    this.coreApiUrl = networkConfig.url;
   }
 
+  static fromName = (networkName: StacksNetworkName): IStacksNetwork => {
+    switch (networkName) {
+      case 'mainnet':
+        return new StacksMainnet();
+      case 'testnet':
+        return new StacksTestnet();
+      default:
+        throw new Error(
+          `Invalid network name provided. Must be one of the following: ${StacksNetworks.join(
+            ', '
+          )}`
+        );
+    }
+  };
```

Comment:
One of the gripes with stacks.js picked up on in the past has been the inconsistent mix between OO and FP styles. Network already uses class style, so adding this static function just follows this style.

Though one thing to consider is the serialisability of these objects. In `@stacks/connect` we pass a network object --> the wallet via a custom event/postMessage comms (which auto-serialises it). We've ran into bugs with this, and also makes the types less useful (there is no type for the result of `JSON.serialize(new StacksMainnet())`)

No action necessary here, but good to be aware of. Perhaps this could also work as a separate fn definition?

Reply:
Good point! Generally, I am trying to move to a more functional approach, especially when composition is useful.

Yes, here we already have classy code. 😉 Technically the new methods are `static` so we could separate them. Yet, it's also nice for developers to be able to benefit from the discoverability of typing `StacksNetwork.` to auto-complete builder-style methods. In general, I am also thinking about namespaces or similar TS techniques to group together functional methods — to increase discoverability.

Reply:
For example, I like the way `clarinet` does this for [`types.`](https://github.com/hirosystems/clarinet/blob/8254f51ed33faa280ff5c7297b2312c2a4f5e9e6/deno/index.ts#L273)